src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgrade.cs

```diff
@@ -76,9 +86,32 @@ public async Task RunAsync(CancellationToken token)
             {
                 // Do not pass the same token as it may have been canceled and we still need to persist this.
                 await _stateManager.SaveStateAsync(context, default);
+                await WriteUpgradeReport(context, token);
```

Comment:
The statement above this is not passing `token` because it may be canceled. Do you only want this to write if in a non-canceled state? (probably fine, just checking). If so, worth adding a comment why it's different than line 88